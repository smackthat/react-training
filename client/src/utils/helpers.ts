export function currencySum(a: number, b: number) {
  return Math.round(a * b * 100) / 100;
}
