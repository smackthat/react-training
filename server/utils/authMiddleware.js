const jwt = require('jsonwebtoken');
const StatusCodes = require('http-status-codes').StatusCodes;

const authMiddlware = (req, res, next) => {

  if (req.originalUrl.includes('api/auth')) {
    next();
  }
  else {
    const authorization = req.get('authorization');

    if (authorization && authorization.toLowerCase().startsWith('bearer')) {
      const token = authorization.substring(7);
  
      try {
        jwt.verify(token, process.env.AUTHSECRET);
        next();
      } catch (error) {
        res.sendStatus(StatusCodes.FORBIDDEN);
      }
    }
    else {
      res.sendStatus(StatusCodes.UNAUTHORIZED);
    }
  }

};

module.exports = authMiddlware;