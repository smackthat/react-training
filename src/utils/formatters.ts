/** Currency formatter */
export function CurrencyFormatter(): Intl.NumberFormat {
  return new Intl.NumberFormat(undefined, {
    style: 'currency',
    currency: 'EUR',
  });
}
