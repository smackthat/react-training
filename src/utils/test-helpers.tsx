import { I18nextProvider } from 'react-i18next';
import React, { ReactElement } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import i18n from 'locales/i18nForTests';
import { configureAppStore } from 'store/configureStore';
import { RootState } from 'types/RootState';
import { EnhancedStore } from '@reduxjs/toolkit';
import { HelmetProvider } from 'react-helmet-async';

// See: https://testing-library.com/docs/react-testing-library/setup

const defaultState: RootState = {
  productsState: {
    loading: false,
    error: false,
    products: [],
  },
  userState: {
    user: null,
    loading: false,
    error: null,
  },
};

function customRender(
  ui: ReactElement,
  {
    initialState = defaultState,
    store = configureAppStore(initialState),
    ...options
  }: { initialState?: RootState; store?: EnhancedStore } = {},
) {
  return {
    result: render(
      <I18nextProvider i18n={i18n}>
        <HelmetProvider>
          <Provider store={store}>{ui}</Provider>
        </HelmetProvider>
      </I18nextProvider>,
      options,
    ),
    store: store,
  };
}

export * from '@testing-library/react';
export { customRender as render };
