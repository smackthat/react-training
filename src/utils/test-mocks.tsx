import { Product, ProductCategory } from 'types/Product';
import { User, UserState } from 'types/User';

export const mockUser: User = {
  id: 2,
  userName: 'TEST',
};

export const mockLoggedInUserState: UserState = {
  user: mockUser,
  loading: false,
  error: null,
  cart: {
    items: [],
  },
};

export const mockProducts: Product[] = [
  {
    id: 1,
    title: 'Shirt',
    description: 'A nice shirt',
    price: 9.99,
    category: ProductCategory.MENS_CLOTHING,
    image: '',
  },
  {
    id: 2,
    title: 'Skirt',
    description: 'A nice skirt',
    price: 19.99,
    category: ProductCategory.WOMENS_CLOTHING,
    image: '',
  },
  {
    id: 3,
    title: 'Phone X',
    description: 'A nice phone',
    price: 999.99,
    category: ProductCategory.ELECTRONICS,
    image: '',
  },
  {
    id: 1,
    title: 'Golden necklace',
    description: 'A nice necklace',
    price: 99.99,
    category: ProductCategory.JEWELERY,
    image: '',
  },
];
