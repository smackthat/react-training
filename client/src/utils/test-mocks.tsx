import { Product, ProductCategory } from 'types/Product';
import { CartItem, User, UserState } from 'types/User';
import { currencySum } from './helpers';

export const mockUser: User = {
  id: 2,
  userName: 'TEST',
};

export const mockLoggedInUserState: UserState = {
  user: mockUser,
  loading: false,
  error: null,
  cart: {
    items: [],
  },
};

export const mockProducts: Product[] = [
  {
    id: 1,
    title: 'Shirt',
    description: 'A nice shirt',
    price: 9.99,
    category: ProductCategory.MENS_CLOTHING,
    image: '',
  },
  {
    id: 2,
    title: 'Skirt',
    description: 'A nice skirt',
    price: 19.99,
    category: ProductCategory.WOMENS_CLOTHING,
    image: '',
  },
  {
    id: 3,
    title: 'Phone X',
    description: 'A nice phone',
    price: 999.99,
    category: ProductCategory.ELECTRONICS,
    image: '',
  },
  {
    id: 4,
    title: 'Golden necklace',
    description: 'A nice necklace',
    price: 99.99,
    category: ProductCategory.JEWELERY,
    image: '',
  },
];

export const mockCartItems: CartItem[] = [
  {
    productId: mockProducts[0].id,
    title: mockProducts[0].title,
    quantity: 8,
    unitPrice: mockProducts[0].price,
    sum: currencySum(mockProducts[0].price, 8),
  },
  {
    productId: mockProducts[3].id,
    title: mockProducts[3].title,
    quantity: 1,
    unitPrice: mockProducts[3].price,
    sum: currencySum(mockProducts[3].price, 1),
  },
];
