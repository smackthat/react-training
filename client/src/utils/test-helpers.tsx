import { I18nextProvider } from 'react-i18next';
import React, { ReactElement } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import i18nForTests from 'locales/i18nForTests';
import { configureAppStore } from 'store/configureStore';
import { RootState } from 'types/RootState';
import { EnhancedStore } from '@reduxjs/toolkit';
import { HelmetProvider } from 'react-helmet-async';
import { InjectedReducersType } from './types/injector-typings';
import { userReducer } from 'app/pages/LoginPage/slice';
import { productsReducer } from 'app/pages/HomePage/slice';
import { i18n } from 'locales/i18n';
import { mockProducts } from './test-mocks';

// See: https://testing-library.com/docs/react-testing-library/setup

export const defaultState: RootState = {
  productsState: {
    loading: false,
    error: false,
    products: mockProducts,
  },
  userState: {
    user: null,
    loading: false,
    error: null,
  },
};

const reducers: InjectedReducersType = {
  userState: userReducer,
  productsState: productsReducer,
};

async function customRender(
  ui: ReactElement,
  {
    initialState = defaultState,
    store = configureAppStore(initialState, reducers),
    ...options
  }: { initialState?: RootState; store?: EnhancedStore } = {},
) {
  const t = await i18n;

  return {
    result: render(
      <I18nextProvider i18n={i18nForTests}>
        <HelmetProvider>
          <Provider store={store}>{ui}</Provider>
        </HelmetProvider>
      </I18nextProvider>,
      options,
    ),
    store: store,
    t: t,
  };
}

export * from '@testing-library/react';
export { customRender as render };
