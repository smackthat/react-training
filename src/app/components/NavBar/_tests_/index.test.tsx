import React from 'react';
import { render } from 'utils/test-helpers';
import { Navbar } from '..';
import { screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';

const renderNavbar = () =>
  render(
    <MemoryRouter>
      <Route path="/">
        <Navbar></Navbar>,
      </Route>
    </MemoryRouter>,
  );

describe('Navbar', () => {
  it('should match snapshot', async () => {
    const navbar = await renderNavbar();
    screen.debug();
    expect(navbar.result).toMatchSnapshot();
  });
});
