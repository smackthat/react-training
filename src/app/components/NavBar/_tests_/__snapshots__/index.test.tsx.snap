// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c4 {
  padding-left: 1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 59px;
  margin-right: 1em;
  width: 22vw;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  color: inherit;
}

.c2 {
  color: #ffffff;
}

.c7 {
  color: #ffffff;
}

.c1 {
  width: 75vw;
  margin: 0 auto;
  padding: 0 1.5rem;
  box-sizing: content-box;
}

.c6 {
  color: white;
  border: 2px solid white;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-size: smaller;
  padding: 0.5em;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-left: 2em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  float: right;
}

.c0 {
  height: 15vh;
  width: 100vw;
  background: -moz-linear-gradient(top,#142062 0%,#004ba8 72%);
  background: -webkit-gradient( linear, left top, left bottom, color-stop(0%,#142062), color-stop(72%,#004ba8) );
  background: -webkit-linear-gradient(top,#142062 0%,#004ba8 72%);
  background: -o-linear-gradient(top,#142062 0%,#004ba8 72%);
  background: -ms-linear-gradient(top,#142062 0%,#004ba8 72%);
  background: linear-gradient(to bottom,#142062 0%,#004ba8 72%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  top: 0;
  padding-bottom: 1em;
  z-index: 3;
  position: fixed;
}

@media (max-width:1100px) {
  .c4 {
    width: 40vw;
  }
}

@media (max-width:700px) {
  .c1 {
    width: 90vw;
  }
}

<body>
    <div>
      <header
        class="c0"
      >
        <div
          class="c1"
        >
          <a
            href="/"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root c2 MuiSvgIcon-fontSizeLarge"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
          <div
            class="c3"
          >
            <div
              class="MuiPaper-root c4 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                aria-expanded="false"
                class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasClearIcon"
                role="combobox"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                    data-shrink="false"
                    for="searchBarAutocomplete"
                    id="searchBarAutocomplete-label"
                  >
                    header.actions.search
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
                      id="searchBarAutocomplete"
                      spellcheck="false"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="Clear"
                        class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                        tabindex="-1"
                        title="Clear"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <button
                aria-label="search"
                class="MuiButtonBase-root MuiIconButton-root c5"
                tabindex="0"
                type="submit"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <a
              class="c6"
              href="/login"
            >
              header.actions.login
            </a>
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root c7"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
      ,
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
    >
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiMenu-paper MuiPopover-paper MuiPaper-elevation8 MuiPaper-rounded"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiMenu-list MuiList-padding"
          role="menu"
          tabindex="-1"
        >
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            role="menuitem"
            tabindex="0"
          >
            languages.english
            <span
              class="MuiTouchRipple-root"
            />
          </li>
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            role="menuitem"
            tabindex="-1"
          >
            languages.finnish
            <span
              class="MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-fznyAO byjxMY"
    >
      <div
        class="PageWrapper-sc-13dr2o1-0 pBcEs"
      >
        <a
          href="/"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root sc-AxiKw hNqCbN MuiSvgIcon-fontSizeLarge"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </a>
        <div
          class="sc-fzplWN bMKMrs"
        >
          <div
            class="MuiPaper-root sc-AxjAm bVQFIM MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              aria-expanded="false"
              class="MuiAutocomplete-root MuiAutocomplete-fullWidth MuiAutocomplete-hasClearIcon"
              role="combobox"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                  for="searchBarAutocomplete"
                  id="searchBarAutocomplete-label"
                >
                  header.actions.search
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                >
                  <input
                    aria-autocomplete="list"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
                    id="searchBarAutocomplete"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Clear"
                      class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
                      tabindex="-1"
                      title="Clear"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <button
              aria-label="search"
              class="MuiButtonBase-root MuiIconButton-root sc-AxirZ kNngH"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <a
            class="sc-fzpans byyTXo"
            href="/login"
          >
            header.actions.login
          </a>
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root sc-AxhCb QBOEV"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
